src/Pos/DB/Block.hs

```diff
@@ -221,14 +224,21 @@ putBi k v = rocksPutBi k v =<< getBlockIndexDB
 delete :: (MonadDB m) => ByteString -> m ()
 delete k = rocksDelete k =<< getBlockIndexDB
 
-getData ::  (MonadIO m, Bi v) => FilePath -> m (Maybe v)
-getData = undefined
+getData ::  (MonadIO m, MonadCatch m, Bi v) => FilePath -> m (Maybe v)
+getData fp = liftIO (decodeMaybe . BSL.fromStrict <$> BS.readFile fp) `catch` handle
+  where
+    decodeMaybe x = case decodeFull x of
```

Comment:
There is [`rightToMaybe`](https://github.com/serokell/universum/blob/f2ccf8afd862e37ccd204c0ef9efde48a05c2d29/src/Monad/Either.hs#L29) in `universum`